runtime/tr.source/trj9/x/codegen/J9CodeGenerator.cpp

```diff
@@ -398,3 +402,71 @@ J9::X86::CodeGenerator::suppressInliningOfRecognizedMethod(TR::RecognizedMethod
       }
    }
 
+void
+J9::X86::CodeGenerator::reserveNTrampolines(int32_t numTrampolines)
+   {
+   TR_J9VMBase *fej9 = (TR_J9VMBase *)(self()->fe());
+   TR::Compilation *comp = self()->comp();
+
+   // Don't do anything if method trampolines are not needed
+   if (!fej9->needsMethodTrampolines())
+      return;
+
+   bool hadClassUnloadMonitor;
+   bool hadVMAccess = fej9->releaseClassUnloadMonitorAndAcquireVMaccessIfNeeded(comp, &hadClassUnloadMonitor);
+
+   TR::CodeCache *curCache = comp->getCurrentCodeCache();
+   TR::CodeCache *newCache = curCache;
+   OMR::CodeCacheErrorCode::ErrorCode status = OMR::CodeCacheErrorCode::ERRORCODE_SUCCESS;
+
+   TR_ASSERT(curCache->isReserved(), "assertion failure");
```

Comment:
This assertion message is not very useful. While we are in this area of code could we improve the message?